docs/specs/native_histograms.md

```diff
@@ -1933,10 +1933,17 @@ the minimum observation will often be in the lowest bucket.
 `histogram_quantile` treats observations of value `NaN` (which SHOULD NOT
 happen, see [above](#special-cases-of-observed-values)) effectively as
 observations of `+Inf`. This follows the rationale that `NaN` is never less
-than any value that `histogram_quantile` returns and is consistent with how
-classic histograms usually treat `NaN` observations (which end up in the `+Inf`
-bucket in most implementations). (TODO: The correct implementation of this
-behavior still needs to be verified by tests.)
+than any value that `histogram_quantile` returns. As long as the quantile
+falls into an existing bucket we return the quantile calculated as if `NaN`
+observations were observed as `+Inf` and also issue an info level annotation
+to let the user know that results are skewed due to `NaN`. This is consistent
+with how classic histograms usually treat `NaN` observations (which end up in
+the `+Inf` bucket in most implementations). If the quantile falls above all
+existing buckets, we return `NaN` due to the fact that we want to be consistent
+with `histogram_fraction`, since `histogram_fraction` considers `NaN`
```

Comment:
Maybe turn this comma into a semicolon or period? It's hard to dissect this sentence currently.

Reply:
I've started to rewrite it and ended up adding a simple and a detailed explanation, please recheck.